src/DependencyInjection/Compiler/AliasDeprecatedPublicServicesCompilerPass.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\DependencyInjection\Compiler;
+
+use Sonata\AdminBundle\Util\BCDeprecationParameters;
+use Symfony\Component\DependencyInjection\Compiler\AbstractRecursivePass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @TODO: Remove it when dropping support for Symfony 4.4 and use "container.private" tag instead.
+ * This file is copied from Symfony since it was added in Symfony 5.1 and we are supporting 4.4
```

Comment:
If we are going to remove it `4.x`, please replace this with a `NEXT_MAJOR: Remove this file.` comment.